site/content/api/Device-Connection-API.md

```diff
@@ -0,0 +1,148 @@
+<!--more-->
+
+The Device Connection API is defined by means of AMQP 1.0 message exchanges, i.e. a client needs to connect to Hono using an AMQP 1.0 client in order to invoke operations of the API as described in the following sections.
+
+# Preconditions
+
+The preconditions for performing any of the operations are as follows:
+
+1. Client has established an AMQP connection with Hono.
+2. Client has established an AMQP link in role *sender* with Hono using target address `device_con/${tenant_id}`. This link is used by the client to send commands concerning device connections to Hono.
+3. Client has established an AMQP link in role *receiver* with Hono using source address `device_con/${tenant_id}/${reply-to}` where *reply-to* may be any arbitrary string chosen by the client. This link is used by the client to receive responses to the device state requests it has sent to Hono. This link's source address is also referred to as the *reply-to* address for the request messages.
+
+This flow of messages is illustrated by the following sequence diagram (showing the AMQP performatives):
+
+![Device_Connection message flow preconditions](../connectToDeviceConnection.png)
+
+# Mandatory Operations
+
+The operations described in the following sections are invoked by Hono's components during run time and are therefore mandatory to implement.
+
+## Set Last Known Gateway for Device
+
+Clients use this command to *set* the gateway that last acted on behalf of a given device. 
+
+The association between device and its last known gateway is used for scenarios where devices with multiple potential gateways are used, along with gateways subscribing to command messages only using their gateway id. In such scenarios, the value set here is needed to route command messages to the right gateway.
+
+This operation is *mandatory* to implement.
+
+**Message Flow**
+
+
+The following sequence diagram illustrates the flow of messages involved in a *Client* retrieving the last known gateway for a device.
+
+![Set Last Known Gateway for Device message flow](../setLastKnownGatewayForDevice_Success.png)
+
+
+**Request Message Format**
+
+The following table provides an overview of the properties a client needs to set on a message to retrieve the last known gateway for a device in addition to the [Standard Request Properties]({{< relref "#standard-request-properties" >}}).
+
+| Name         | Mandatory | Location                 | AMQP Type | Description |
+| :----------- | :-------: | :----------------------- | :-------- | :---------- |
+| *subject*    | yes       | *properties*             | *string*  | MUST be set to `setLastKnownGatewayForDevice`. |
+| *gateway_id* | yes       | *application-properties* | *string*  | The identifier of the gateway that last acted on behalf of a given device. If a device connects directly instead of through a gateway, the device identifier itself is to be used here. |
```

Comment:
> The identifier of the gateway that last acted on behalf of the device identified by the *device_id* property. If a device connects directly instead of through a gateway, the device's identifier MUST be specified here.